src/Fields/set!.jl

```diff
@@ -1,3 +1,4 @@
+using CUDAnative
```

Comment:
do we need `@hascuda` in front of this?

Reply:
Hmmm, I think in the past we did but some versions ago all the CUDA packages became safe to import without a GPU. I guess we can clean up some uses of `@hascuda`.

Either way, tests are chugging along fine.

Reply:
Not quite:

```
julia> using CUDAnative
[ Info: CUDAdrv.jl failed to initialize, GPU functionality unavailable (set JULIA_CUDA_SILENT or JULIA_CUDA_VERBOSE to silence or expand this message)
```

Reply:
Ah good point, I forgot about that warning (or info message). Hmmm, the fact that I've seen it probably means that there's some CUDA package import that needs a `@hascuda`.

Well, one approach is to `@hascuda` everything that might need it. Another is to show that warning to everyone.

Reply:
I vote `@hascuda`. We don't *require* CUDA, we simply support it. (It'd still be nice to emit an interpretable error if somebody tries to use `architecture=GPU()` without CUDA though.)

Reply:
:+1: We already do:
https://github.com/climate-machine/Oceananigans.jl/blob/4ef4d11eb95c2356847aeb4386e7e15e8d19fc9e/src/Models/incompressible_model.jl#L86-L88